packages/admin-client/src/common/components/NotifyButton/NotifyButton.tsx

```diff
@@ -1,28 +1,79 @@
+import { CatchmentContext } from "../../contexts/CatchmentContext/CatchmentContext"
 import { faFlag } from "@fortawesome/pro-solid-svg-icons"
 import { COLOURS } from "../../../Colours"
 import { Button } from "@mui/material"
 
 const NotifyButton: React.FC = () => {
-    // TODO: this variable should correspond to a value persisted in the user's settings.
-    const [notificationsOn, setNotificationsOn] = useState(true)
+    const cc = useContext(CatchmentContext)
+    const dataProvider = useDataProvider()
+    const listContext = useListContext()
+    const [notifications, setNotifications] = useState(false)
+    const [loading, setLoading] = useState(true)
     const [ariaLabel, setAriaLabel] = useState("Turn off notifications")
 
-    const handleClick = (event) => {
-        setNotificationsOn(!notificationsOn)
+    const handleClick = () => {
+        setLoading(true)
+        if (notifications === true) {
+            dataProvider
+                .deleteNotifications(listContext.resource.split("s")[0], { catchment: cc.catchment.id })
+                .then((response: any) => {
+                    console.log(response)
+                    setLoading(false)
+                    setNotifications(false)
+                })
+                .catch((error: any) => {
+                    setLoading(false)
+                    console.log(error)
+                    alert("There was an error turning off notifications.")
+                })
+        } else {
+            dataProvider
+                .addNotifications(listContext.resource.split("s")[0], { catchment: cc.catchment.id })
+                .then((response: any) => {
+                    console.log(response)
+                    setLoading(false)
+                    setNotifications(true)
+                })
+                .catch((error: any) => {
+                    setLoading(false)
+                    console.log(error)
+                    alert("There was an error turning on notifications.")
+                })
+        }
     }
 
     useEffect(() => {
-        setAriaLabel("Turn " + (notificationsOn ? "off" : "on") + " notifications")
-    }, [notificationsOn])
+        setAriaLabel("Turn " + (notifications ? "off" : "on") + " notifications")
+    }, [notifications])
+
+    // On first load or catchment change, check if the user has notifications enabled for this catchment.
+    useEffect(() => {
+        setLoading(true)
+        dataProvider
+            .getNotifications(listContext.resource.split("s")[0], { catchment: cc.catchment.id })
+            .then((response: any) => {
+                setLoading(false)
+                if (response.length > 0) {
+                    setNotifications(true)
+                } else {
+                    setNotifications(false)
+                }
+            })
+    }, [cc.catchment.id])
+
+    if (loading) {
+        return <LoadingIndicator sx={{ color: COLOURS.MEDIUMGREY, maxWidth: "300px", marginRight: "4em" }} />
```

Comment:
Loading indicator is animated during first page load, but static when button is clicked. Unclear why, may take some investigation. May deserve its own ticket. Otherwise seems to be working!